packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,759 @@
+                        width: widget.preferredWidth,
+                        extendedWidth: widget.extendedWidth,
+                        extendedTransitionAnimation: _extendedAnimation,
+                        selected: widget.currentIndex == i,
+                        icon: widget.currentIndex == i ? widget.destinations[i].activeIcon : widget.destinations[i].icon,
+                        label: widget.destinations[i].label,
+                        destinationAnimation: _destinationAnimations[i],
+                        labelType: labelType,
+                        iconTheme: widget.currentIndex == i ? selectedIconTheme : unselectedIconTheme,
+                        labelTextStyle: widget.currentIndex == i ? selectedLabelTextStyle : unselectedLabelTextStyle,
+                        onTap: () {
+                          widget.onDestinationSelected(i);
+                        },
+                        indexLabel: localizations.tabLabel(
+                          tabIndex: i + 1,
+                          tabCount: widget.destinations.length,
+                        ),
+                      ),
+                  ],
+                ),
+              ),
+              if (widget.trailing != null)
+                if (_extendedAnimation.value > 0)
+                  SizedBox(
+                    width: lerpDouble(widget.preferredWidth, widget.extendedWidth, _extendedAnimation.value),
+                    child: widget.trailing,
+                  )
+                else
+                  widget.trailing,
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  MainAxisAlignment _resolveGroupAlignment(NavigationRailGroupAlignment groupAlignment) {
+    switch (groupAlignment) {
+      case NavigationRailGroupAlignment.top:
+        return MainAxisAlignment.start;
+      case NavigationRailGroupAlignment.center:
+        return MainAxisAlignment.center;
+      case NavigationRailGroupAlignment.bottom:
+        return MainAxisAlignment.end;
+    }
+    return MainAxisAlignment.start;
+  }
+
+  void _disposeControllers() {
+    for (final AnimationController controller in _destinationControllers)
+      controller.dispose();
+    _extendedController.dispose();
+  }
+
+  void _initControllers() {
+    _destinationControllers = List<AnimationController>.generate(widget.destinations.length, (int index) {
+      return AnimationController(
+        duration: kThemeAnimationDuration,
+        vsync: this,
+      )..addListener(_rebuild);
+    });
+    _destinationAnimations = _destinationControllers.map((AnimationController controller) => controller.view).toList();
+    _destinationControllers[widget.currentIndex].value = 1.0;
+    _extendedController = AnimationController(
+      duration: kThemeAnimationDuration,
+      vsync: this,
+
+    );
+    _extendedAnimation = CurvedAnimation(
+      parent: _extendedController,
+      curve: Curves.easeInOut,
+    );
+    _extendedController.addListener(() {
+      setState(() {
```

Comment:
You can use your `_rebuild` helper here, though I'm not sure if there's much value in even having the helper

Reply:
Done. I took that from bottom navigation bar as well. I think it helps explain its use because usually empty setStates are discouraged